multus/multus.go

```diff
@@ -417,23 +426,30 @@ func cmdDel(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) err
 
 	numK8sDelegates, kc, err := k8s.TryLoadPodDelegates(k8sArgs, in, kubeClient)
 	if err != nil {
-		return err
+		// in error case, need to proceed without return for bailout
+		_ = logging.Errorf("Multus: Err in getting k8s network from pod: %v", err)
 	}
 
 	if numK8sDelegates == 0 {
+		logging.Debugf("cmdDel: reading delegates from cache")
 		// re-read the scratch multus config if we have only Multus delegates
 		netconfBytes, err := consumeScratchNetConf(args.ContainerID, in.CNIDir)
 		if err != nil {
 			if os.IsNotExist(err) {
 				// Per spec should ignore error if resources are missing / already removed
 				return nil
 			}
-			return logging.Errorf("Multus: Err in  reading the delegates: %v", err)
+			return logging.Errorf("Multus: Err in reading the delegates: %v", err)
 		}
 
 		if err := json.Unmarshal(netconfBytes, &in.Delegates); err != nil {
 			return logging.Errorf("Multus: failed to load netconf: %v", err)
 		}
+	} else {
```

Comment:
@s1061123 this will delete the cache before actually calling the plugins, which leaves the possibility of errors being returned after the cache files are gone. I think we should move the cache deletion to the end of the function, after all other calls have run.

Reply:
Fixed.